Python/optimizer.c

```diff
@@ -154,13 +154,13 @@ PyUnstable_GetOptimizer(void)
 }
 
 static _PyExecutorObject *
-make_executor_from_uops(_PyUOpInstruction *buffer, const _PyBloomFilter *dependencies);
+make_executor_from_uops(_PyUOpInstruction *buffer, int length, const _PyBloomFilter *dependencies);
 
 static int
 init_cold_exit_executor(_PyExecutorObject *executor, int oparg);
 
 static int cold_exits_initialized = 0;
-static _PyExecutorObject COLD_EXITS[UOP_MAX_TRACE_LENGTH] = { 0 };
+static _PyExecutorObject COLD_EXITS[UOP_MAX_TRACE_LENGTH/4] = { 0 };
```

Comment:
Ah, so *this* is where your memory savings comes from! Cutting 512 side exits down to 200. :)

Is 4 plucked out of thin air?

Reply:
Not quite. I think that the maximum number of exits possible is 1/6 of `UOP_MAX_TRACE_LENGTH` due to `_CHECK_VALIDITY/_SET_IP`, the need to evaluate the values for exit tests, and the additional exit handling instruction at the end. 1/4 is a safe over estimate.